discord.go

```diff
@@ -153,13 +190,22 @@ func discordMessageHandler(_ *discordgo.Session, m *discordgo.MessageCreate) {
 	runCommands(msgContent, DiscordUser)
 }
 
-var cacheSize = 20
+const cacheSize = 20
 
-// basic cache system
-var imageCache = make([]struct {
-	user  string
-	image string
-}, cacheSize)
+var imageCache = make(imgCache, cacheSize)
+
+type imgCache map[string]string
+
+func (i *imgCache) add(user, image string) {
+	if len(*i) >= cacheSize {
+		// remove the first value
+		for k := range *i {
+			delete(*i, k)
+			break
+		}
+	}
+	(*i)[user] = image
+}
```

Comment:
I don't think you need to delete the previous value in a map. You can set it directly.

Reply:
i am trying to keep the map from getting to large, as the images themselves are stored in the map and users can easily change color or name and thus create extra entries in the cache. For that reason i believe there should be a limit as to not allow the ram usage to increase that much.

Reply:
Having the cache linked to a list of online members would be better but would also mean deeper changes into devzat's code that is not related to the discord bridge.

Reply:
ah nvm I see what the code does. we should make this an LRU cache.